backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisService.java

```diff
@@ -278,7 +287,37 @@ public AnalysisConfiguration getAnalysisConfigForEngine(String configScopeId, Li
   }
 
   public void setUserAnalysisProperties(String configScopeId, Map<String, String> properties) {
+    boolean reAnalyseCAndCppFiles = shouldReAnalyseCFiles(configScopeId, properties);
     userAnalysisPropertiesRepository.setUserProperties(configScopeId, properties);
+    if (reAnalyseCAndCppFiles) {
+      var settings = configurationRepository.getEffectiveBinding(configScopeId)
+        .map(effectiveBinding -> storageService.binding(effectiveBinding)
+          .analyzerConfiguration().read().getSettings().getAll())
+        .orElseGet(Map::of);
+      var languageDetection = new LanguageDetection(new MapSettings(settings).asConfig());
+      var openFiles = openFilesRepository.getOpenFilesForConfigScope(configScopeId);
+
+      var openCOrCppFiles = openFiles.stream()
+        .filter(file -> {
+          var clientFile = fileSystemService.getClientFile(file);
+          var clientForcedLanguage = clientFile != null ? clientFile.getDetectedLanguage(): null;
+          var detectedLanguage = clientForcedLanguage != null ? clientForcedLanguage: languageDetection.language(file);
+          return detectedLanguage == SonarLanguage.C || detectedLanguage == SonarLanguage.CPP;
+        })
+        .collect(toList());
+      if (!openFiles.isEmpty()) {
+        triggerAnalysis(configScopeId, openCOrCppFiles);
+      }
```

Comment:
I wonder if we shouldn't keep it simple and reanalyze all open files. In most cases, I would assume in a C or C++ project, almost all open files would need to be re-analyzed

Reply:
Changed implementation. Removed filtering. All open files will be analyzed now.